writeengine/bulk/we_bulkloadbuffer.cpp

```diff
@@ -1721,107 +1690,1381 @@ int BulkLoadBuffer::parseCol(ColumnInfo& columnInfo)
+          pVal = &llDate;
+          memcpy(p, pVal, width);
+        }
+      }
+      else
+      {
+        // datetime conversion here
+        // default column type is TIMESTAMP
+        std::shared_ptr<arrow::TimestampArray> timeArray = std::static_pointer_cast<arrow::TimestampArray>(columnData);
+        for (unsigned int i = 0; i < cbs; i++)
+        {
+          int rc = 0;
+          void *p = buf + i * width;
+          if (columnData->IsNull(i))
+          {
+            if (column.fWithDefault)
+            {
+              llDate = column.fDefaultInt;
+            }
+            else
+            {
+              llDate = joblist::DATETIMENULL;
+              pVal = &llDate;
+              memcpy(p, pVal, width);
+              continue;
+            }
+          }
+          else
+          {
+            int64_t timeVal = timeArray->Value(i);
+            llDate = dataconvert::DataConvert::convertArrowColumnDatetime(timeVal, rc);
+          }
+          if (rc == 0)
+          {
+            if (llDate < bufStats.minBufferVal)
+              bufStats.minBufferVal = llDate;
+
+            if (llDate > bufStats.maxBufferVal)
+              bufStats.maxBufferVal = llDate;
+          }
+          else
+          {
+            llDate = 0;
+            bufStats.satCount++;
+          }
+          pVal = &llDate;
+          memcpy(p, pVal, width);
+        }
+
+      }
+      break;
+    }
+
+    //----------------------------------------------------------------------
+    // WIDE DECIMAL
+    //----------------------------------------------------------------------
+    case WriteEngine::WR_BINARY:
+    {
+      // Parquet does not have data type with 128 byte
+      std::shared_ptr<arrow::Decimal128Array> decimalArray = std::static_pointer_cast<arrow::Decimal128Array>(columnData);
+      std::shared_ptr<arrow::DecimalType> fType = std::static_pointer_cast<arrow::DecimalType>(decimalArray->type());
+      const int128_t* dataPtr = decimalArray->data()->GetValues<int128_t>(1);
+
+
+      for (unsigned int i = 0; i < cbs; i++)
+      {
+        void* p = buf + i * width;
+        bool bSatVal = false;
+        if (columnData->IsNull(i))
+        {
+          if (!column.autoIncFlag)
+          {
+            if (column.fWithDefault)
+            {
+              bigllVal = column.fDefaultWideDecimal;
+            }
+            else
+            {
+              bigllVal = datatypes::Decimal128Null;
+              pVal = &bigllVal;
+              memcpy(p, pVal, width);
+              continue;
+            }
+          }
+          else
+          {
+            bigllVal = fAutoIncNextValue++;
+          }
+        }
+        else
+        {
+          // compare parquet data precision and scale with table column precision and scale
+          
+          // Get int and frac part
+          memcpy(&bigllVal, dataPtr + i, sizeof(int128_t));
+
+
+        }
+        if (bSatVal)
+          bufStats.satCount++;
```

Comment:
For this one, because the decimal128 data user wants to import should fit its precision and scale in the parquet data, I think the data can't saturate.

Reply:
Ok. Please add a comment explaining this scenario.